src/bin/rga-fzf-open.rs

```diff
@@ -10,10 +10,16 @@ fn main() -> anyhow::Result<()> {
     let query = args.next().context("no query")?;
     let fname = args.next().context("no filename")?;
     // let instance_id = std::env::var("RGA_FZF_INSTANCE").unwrap_or("unk".to_string());
+    use std::env;
 
+    let (cmd, pdf_cmd) = match env::consts::OS {
+        "macos" => ("open", "open -a Preview.app"), // use native Preview for macOs
+        "linux" => ("xdg-open", "evince"),          // use evince for linux
+        &_ => ("", ""),
```

Comment:
I haven't personally verified this, but [`start`](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/start) appears to be relevant command on Windows. Supposedly when passed a file as its sole argument it behaves the same as double-clicking the file in explorer.exe, namely opening the file in the associated default program.

Reply:
Thanks for the info! I did some digging around that and apparently `start` [doesn’t work consistently across shells in Windows](https://superuser.com/questions/246825/open-file-from-the-command-line-on-windows). This info might be outdated since the comment talking about this is from 2018. 

Also, in the doc you sent about `start` it says you can run non-executable files through their file association by typing the name of the file as a command. A little bit of search led me to this two articles from last year which claim the same thing: [howtogeek article](https://www.howtogeek.com/674537/how-to-find-and-open-files-using-command-prompt/), [windows loop article](https://windowsloop.com/open-files-with-command-prompt/).

I might try to personally test these and see which alternative is best with a Windows VM or a Windows Docker container.

Reply:
> Also, in the doc you sent about `start` it says you can run non-executable files through their file association by typing the name of the file as a command.

Thinking out loud: I wonder if that is implemented at the level of cmd.exe/PowerShell or somewhere deeper. If the former, it might be easier to look at how other Rust programs are solving this than trying to reason it from first principles…

Reply:
Finally researched this a bit more. It looks like `start` is a built-in in both cmd.exe and Windows variants of PowerShell, _not_ a standalone tool that can be trivially executed. The standard approach (see [CPython's implementation](https://github.com/python/cpython/commit/f58a7aafea13f9214065bf5d97855cfce154d1c3#diff-b5b7e4f5599916b2018e911ef423f9d3fd46cb71268e1a73c74a1c9ae2cb6791) as an example) seems to be to spin out a shell process to run `start`.

However, it might be worth just pulling in a dependency like [open](https://crates.io/crates/open) rather than implement this directly. There appear to be some odd edge cases that I'd rather not have to keep track of and handle, e.g. <https://github.com/Byron/open-rs/issues/73>.